libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -400,8 +400,10 @@ open class GossipRouter(
     override fun broadcastOutbound(msg: PubsubMessage): CompletableFuture<Unit> {
         msg.topics.forEach { lastPublished[it] = currentTimeSupplier() }
 
+        val floodPublish = msg.protobufMessage.data.size() <= params.floodPublishMaxMessageSizeThreshold
```

Comment:
We already refer to 'message size' in a couple of places. I would add this notion to `PubsubMessage` interface: 
```patch
Index: libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubRouter.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubRouter.kt b/libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubRouter.kt
--- a/libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubRouter.kt	(revision 1defc232bebd9bb3eaafd05be1e7fa14f436f328)
+++ b/libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubRouter.kt	(date 1729512008659)
@@ -22,6 +22,9 @@
     val topics: List<Topic>
         get() = protobufMessage.topicIDsList
 
+    val size: Long
+        get() = protobufMessage.data.size().toLong()
+
     fun messageSha256() = sha256(protobufMessage.toByteArray())
 
     override fun equals(other: Any?): Boolean
```

Reply:
right, yes.. seems like you like the idea to expose `Long` even if under the hood we can't go over 2gb (protobuf limit?)

Reply:
Oh, it may be an overkill. Let's stick with just Int then